## Kaggle Sentiment Analysis for financial news 데이터 셋 load

https://www.kaggle.com/ankurzing/sentiment-analysis-for-financial-news

In [40]:
path = "./archive/all-data.csv"

In [41]:
all_lines = []

In [42]:
import csv

f = open(path, 'r', encoding="ISO-8859-1")
lines = csv.reader(f)


for line in lines:    
    all_lines.append(line)    

In [43]:
all_lines[:8]

[['neutral',
  'According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .'],
 ['neutral',
  'Technopolis plans to develop in stages an area of no less than 100,000 square meters in order to host companies working in computer technologies and telecommunications , the statement said .'],
 ['negative',
  'The international electronic industry company Elcoteq has laid off tens of employees from its Tallinn facility ; contrary to earlier layoffs the company contracted the ranks of its office workers , the daily Postimees reported .'],
 ['positive',
  'With the new production plant the company would increase its capacity to meet the expected increase in demand and would improve the use of raw materials and therefore increase the production profitability .'],
 ['positive',
  "According to the company 's updated strategy for the years 2009-2012 , Basware targets a long-term net sales growth in the range of 20 % -40 % with an

In [4]:
#데이터 셋의 문장 개수
len(all_lines)

4846

In [5]:
all_lines[0]

['neutral',
 'According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .']

In [6]:
#데이터셋은 neutral, negative, postive 으로 labelling 되어있음
all_lines[0][0]

'neutral'

In [7]:
# 데이터셋의 문장 부분
all_lines[0][1]

'According to Gran , the company has no plans to move all production to Russia , although that is where the company is growing .'

## 네이버 파파고 API활용 번역
4846개 문장을 번역한 후 제대로 번역되지 않은 문장은 직접 수정

In [8]:
import os
import sys
import urllib.request
import json

In [9]:
#client_id와 client_secret 정보
client_id = "-"
client_secret = "-"

In [11]:
def translator(text, source='en', target='ko'):  
    
    Text = urllib.parse.quote(text)    
    data = f'source={source}&target={target}&text={Text}'
    
    url = "https://naveropenapi.apigw.ntruss.com/nmt/v1/translation"
    request = urllib.request.Request(url)
    request.add_header("X-NCP-APIGW-API-KEY-ID",client_id)
    request.add_header("X-NCP-APIGW-API-KEY",client_secret)
    response = urllib.request.urlopen(request, data=data.encode("utf-8"))
    rescode = response.getcode()
    if(rescode==200):
        response_body = response.read()
        response = json.loads(response_body.decode('utf-8'))    
        result = response['message']['result']['translatedText']
        return(result)
    else:
        return ("Error")       
        

In [16]:
# #API 테스트 문장
test = "the broker gave an `` outperform '' recommendation on the stock ."
translator(test)

'주식 중개인은 주식에 대해 "상당히" 추천했다.'

In [17]:
done = []

In [18]:
for i in range(len(all_lines)):     
    input = all_lines[i][1]
    result = translator(input)
    done.append(result)   

In [23]:
len(done)

4846

In [25]:
for i in range(len(done)):
    temp = done[i]
    all_lines[i].append(temp)

In [27]:
len(all_lines)

4846

## 엑셀로 내보내기
리스트 형식의 데이터를 엑셀로 내보냈습니다. 

In [28]:
from openpyxl import Workbook

In [31]:
wb = Workbook()

In [34]:
sheet = wb.active

In [35]:
for i in range(len(all_lines)):
    sheet.append(all_lines[i])

In [36]:
wb.save('Kaggle_translated.xlsx')